core/domain/collection_domain.py

```diff
@@ -230,13 +263,54 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id)
 
 
+class CollectionDict(TypedDict, total=False):
+    """Type for the collection object"""
+
+    id: str
+    title: str
+    category: str
+    objective: str
+    language_code: str
+    tags: List[str]
+    schema_version: int
+    nodes: list[CollectionNodeDict]
+    version: int
+    created_on: Union[datetime.datetime, str, None]
+    last_updated: Union[datetime.datetime, str, None]
```

Comment:
Why this `Union`?

Reply:
In collection's construction function (__init__), we expecting only 2 values none and datetime object. when we use `serialize` method to serialize collection object, the datetime object is also converted to string.
https://github.com/oppia/oppia/blob/add62e844665ab43229dbfd3cd207dc0e0343356/core/domain/collection_domain.py#L406

Reply:
Can you somehow refactor it to only use either `datetime` or `str` (`None` is fine)?

Reply:
I tried to refactor it by initiallizing the `created_on` and `last_updated` parameters with `datetime.datetime.utcnow()` as default value.